src/hashTable.js

```diff
@@ -1,63 +1,94 @@
 class Node{
-  constructor() {
-    this.key = 0,
-    this.data = 0,
+  constructor(key, data) {
+    this.key = key,
+    this.data = data,
     this.next = null
   }
 }
 
 export default class Hashtable{
   constructor(){
-    this._length,
+    this._length = 0,
     this.dataStore = []
   }
 
-  hashFunction(key) {
+  //takes a key and returns a hash number which will be where the hash will live.
+  hashFunction(key){
     let sum = 0
     for (let i = 0; i < key.length; i++) {
       sum += key.charCodeAt(i)
     }
     return sum % 31
   }
+  //get the number of hashes in your hash table.
+  size(){ return this._length }
 
-  chain(){
-    //iterate through the linkedlist until you reach a node whos next is null.
-    //set that next to the new node that is created.
-  }
 
-<<<<<<< Updated upstream
-  put(id, data){
-    let hash = hashFunction(id)
-=======
+  //creates a new hash or chains off an existing hash.
   put(key, data){
+    let newNode = new Node(key, data)
     let hash = this.hashFunction(key)
->>>>>>> Stashed changes
     //see if hash # exists on the hash table.
-    if(!this.dataStore.contains(hash)){
+    if(!this.dataStore[hash]){
       //if it doesnt, create a new hash index in the table and put data there.
-
+      this.dataStore[hash] = newNode
     }else{
       //otherwise chain off the lead hash into a linkedlist.
-<<<<<<< Updated upstream
-      
-=======
-
->>>>>>> Stashed changes
+      this.chain(key, data, hash, newNode)
     }
+    ++this._length
   }
 
-  get(){
+  get(key){
+    let hash = this.hashFunction(key)
     //finds the indexed hash number and looks at the key to locate the correct node.
-    //returns that node's data.
+    if(this.dataStore[hash]){
+      let currentNode = this.dataStore[hash]
+      if(currentNode.key === key){
+        //returns that node's data.
+        return currentNode.data
+      }else{
+        while(currentNode.next){
+          if(currentNode.key === key){
+            //returns that node's data.
+            return currentNode.data
+          }else{
+            currentNode = currentNode.next
+          }
+        }
+        if(currentNode.key === key){
+          //returns that node's data.
+          return currentNode.data
+        }else{
+          return "sorry your data is not here"
+        }
+      }
+    }
   }
 
-<<<<<<< Updated upstream
-  contains(){
-    //returns a boolean that state whether or not the hashtable contains that key.
-=======
+  //creates a linked-list at that designated hash location.
+  chain(key, data, hash, newNode){
```

Comment:
I've got benchmark reviews all morning, and a Dr. appt in the afternoon, so I might not have time.  ☹️   One of the goals of a hash table is to have constant time insertion and retrieval, or, when using the separate chaining approach to collision resolution, as close to constant time as possible.

With the current implementation, you will almost never have constant time insertion *or* retrieval.  If you insert at the head of the linked list, you will at least *always* have constant time insertion, and O( size of list ) for retrieval.  The only time I would *not* do it this way is if the characteristics of the data retrieval were known in advance, and I knew that I would always be removing the oldest element from the linked list, in which case I may as well do the work of insertion at the end, so that I can guarantee constant time retrieval.